src/axom/sidre/tests/sidre_mfem_datacollection.cpp

```diff
@@ -237,6 +237,281 @@ TEST(sidre_datacollection, dc_reload_mesh)
+
+  // Then add the species set
+  const bool volume_dependent = false;
+  sdc.AssociateSpeciesSet("partial_density", "specset", "matset", volume_dependent);
+
+  mfem::GridFunction partial_density_1_1(&fes);
+  partial_density_1_1 = 0.4;
+  sdc.RegisterField("partial_density_001_001", &partial_density_1_1);
+  mfem::GridFunction partial_density_1_2(&fes);
+  partial_density_1_2 = 0.3;
+  sdc.RegisterField("partial_density_001_002", &partial_density_1_2);
+  mfem::GridFunction partial_density_2_1(&fes);
+  partial_density_2_1 = 0.2;
+  sdc.RegisterField("partial_density_002_001", &partial_density_2_1);
+  mfem::GridFunction partial_density_2_2(&fes);
+  partial_density_2_2 = 0.1;
+  sdc.RegisterField("partial_density_002_002", &partial_density_2_2);
+
+  EXPECT_TRUE(sdc.verifyMeshBlueprint());
+
+  const auto bp_grp = sdc.GetBPGroup();
+  EXPECT_TRUE(bp_grp->hasGroup("specsets"));
+  EXPECT_TRUE(bp_grp->hasGroup("specsets/specset"));
+  EXPECT_TRUE(bp_grp->hasView("specsets/specset/volume_dependent"));
+  EXPECT_FALSE(static_cast<axom::int8>(
+    bp_grp->getView("specsets/specset/volume_dependent")->getScalar()));
+  EXPECT_TRUE(bp_grp->hasView("specsets/specset/matset"));
+  EXPECT_EQ(std::string(bp_grp->getView("specsets/specset/matset")->getString()),
+            "matset");
+  checkReferentialEquality(bp_grp,
+                           "matsets/matset/volume_fractions/001",
+                           "fields/volume_fraction_001/values");
+  checkReferentialEquality(bp_grp,
+                           "matsets/matset/volume_fractions/002",
+                           "fields/volume_fraction_002/values");
+  checkReferentialEquality(bp_grp,
+                           "specsets/specset/matset_values/001/001",
+                           "fields/partial_density_001_001/values");
+  checkReferentialEquality(bp_grp,
+                           "specsets/specset/matset_values/001/002",
+                           "fields/partial_density_001_002/values");
+  checkReferentialEquality(bp_grp,
+                           "specsets/specset/matset_values/002/001",
+                           "fields/partial_density_002_001/values");
+  checkReferentialEquality(bp_grp,
+                           "specsets/specset/matset_values/002/002",
+                           "fields/partial_density_002_002/values");
+}
+
+TEST(sidre_datacollection, create_material_dependent_field)
+{
+  // 1D mesh divided into 10 segments
+  mfem::Mesh mesh(10);
+  MFEMSidreDataCollection sdc(testName(), &mesh);
+  mfem::H1_FECollection fec(1, mesh.Dimension());
+  mfem::FiniteElementSpace fes(&mesh, &fec);
+
+  // Add a material set so we can associate the field with it
+  sdc.AssociateMaterialSet("volume_fraction", "matset");
+  mfem::GridFunction vol_frac_1(&fes);
+  vol_frac_1 = 1.0;
+  sdc.RegisterField("volume_fraction_001", &vol_frac_1);
+
+  // Then mark the field as material-dependent
+  sdc.AssociateMaterialDependentField("density", "matset");
```

Comment:
@joshessman-llnl -- could you please add a test where you call the functions in the following order:
```cpp
  sdc.AssociateMaterialDependentField("density", "matset");
  sdc.AssociateMaterialSet("volume_fraction", "matset");
```

Is this an error? Should it be?

Reply:
I think it's reasonable to make this a warning but not an error - it's not fatal (i.e., it wouldn't go on to deref a nullptr or something like that) but is probably an undesirable usage pattern.